example/subelements/basic_quad_refinement.cxx

```diff
@@ -0,0 +1,125 @@
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+#include <sc_refcount.h>
+#include <sc_shmem.h>
+#include <p4est_connectivity.h>
+#include <p8est_connectivity.h>
+#include <t8_schemes/t8_new_feature/t8_subelements_cxx.hxx>
+#include <t8_forest/t8_forest_adapt.h>
+#include <t8_forest.h>
+#include <t8_cmesh_vtk.h>
+#include <t8_vec.h>
+#include <example/common/t8_example_common.h>
+
+typedef struct
+{
+  double              mid_point[3];
+  double              radius;
+} t8_basic_sphere_data_t;
+
+/* Compute the distance to a sphere arount a mid_point with given radius. */
+static double
+t8_basic_level_set_sphere (const double x[3], double t, void *data)
+{
+  t8_basic_sphere_data_t *sdata = (t8_basic_sphere_data_t *) data;
+  double             *M = sdata->mid_point;
+
+  return t8_vec_dist (M, x) - sdata->radius;
+}
+
+static void
+t8_basic_refine_test (t8_eclass_t eclass)
+{
+  t8_forest_t         forest;
+  t8_forest_t         forest_adapt;
+  t8_cmesh_t          cmesh;
+  char                filename[BUFSIZ];
+  int                 initlevel = 1;                
+  int                 minlevel = 1; 
+  int                 maxlevel = 5;
+
+  t8_forest_init (&forest);
+  t8_forest_init (&forest_adapt);
+
+  /* building the cmesh, using the initlevel */
+  cmesh = t8_cmesh_new_hypercube (eclass, sc_MPI_COMM_WORLD, 0, 0, 0);
+  t8_forest_set_cmesh (forest, cmesh, sc_MPI_COMM_WORLD);
+  t8_forest_set_scheme (forest, t8_scheme_new_subelement_cxx ());
+  t8_forest_set_level (forest, initlevel);
+
+  t8_forest_commit (forest);
+
+  snprintf (filename, BUFSIZ, "e_s_forest_uniform_%s",
+            t8_eclass_to_string[eclass]);
+  t8_forest_write_vtk (forest, filename);
+
+  /* adapting the cmesh, using user-data, up to a given max and min level */
+  t8_example_level_set_struct_t ls_data;
+  t8_basic_sphere_data_t sdata;
+
+  sdata.mid_point[0] = 0;
+  sdata.mid_point[1] = 1;
+  sdata.mid_point[2] = 0;
+  sdata.radius = 0.3;
+
+  ls_data.band_width = 2;
+  ls_data.L = t8_basic_level_set_sphere;
+  ls_data.min_level = minlevel;
+  ls_data.max_level = maxlevel;
+  ls_data.udata = &sdata;
+
+  t8_forest_set_user_data (forest_adapt, &ls_data);
+  t8_forest_set_adapt (forest_adapt, forest, t8_common_adapt_level_set, 1);
+  t8_forest_set_balance (forest_adapt, NULL, 0); 
+  #if 1
+  t8_forest_set_subelements (forest_adapt, NULL);
+  #endif
+
+  t8_forest_commit (forest_adapt);
+
+  snprintf (filename, BUFSIZ, "e_s_forest_adapt_%s", t8_eclass_to_string[eclass]);
+  t8_forest_write_vtk (forest_adapt, filename);
+
+  t8_forest_unref (&forest_adapt);
+}
+
+int
+main (int argc, char **argv)
+{
+  int                 mpiret;
+
+  mpiret = sc_MPI_Init (&argc, &argv);
+  SC_CHECK_MPI (mpiret);
+
+  sc_init (sc_MPI_COMM_WORLD, 1, 1, NULL, SC_LP_ESSENTIAL);
+  t8_init (SC_LP_DEFAULT);
+
+  t8_basic_refine_test (T8_ECLASS_QUAD);
```

Comment:
You should change the name of the function to something relating to subelements.
It is ok that you copied the code here, but rename it.

Reply:
done (new name: t8_refine_with_subelements)